hibernate-core/src/main/java/org/hibernate/query/internal/ParameterMetadataImpl.java

```diff
@@ -201,7 +195,12 @@ public Set<QueryParameterImplementor<?>> getRegistrations() {
 
 	@Override
 	public boolean hasAnyMatching(Predicate<QueryParameterImplementor<?>> filter) {
-		for ( QueryParameterImplementor<?> queryParameter : queryParameters.keySet() ) {
+		for (QueryParameterImplementor<?> queryParameter : queryParametersByNames.values()  ){
+			if ( filter.test( queryParameter ) ) {
+				return true;
+			}
+		}
+		for (QueryParameterImplementor<?> queryParameter : queryParametersByLabels.values()  ){
```

Comment:
Why did you change this? Couldn't we just keep the previous implementation with a single loop?